crnlib/crn_threading_pthreads.cpp

```diff
@@ -95,14 +93,16 @@ void mutex::unlock() {
   pthread_mutex_unlock(&m_mutex);
 }
 
-void mutex::set_spin_count(unsigned int count) {
-  count;
+void mutex::set_spin_count(unsigned int /* count */) {
 }
 
+#if !defined(__APPLE__)
+semaphore::semaphore(long initialCount, long /* maximumCount */, const char* /* pName */) {
+#else
 semaphore::semaphore(long initialCount, long maximumCount, const char* pName) {
+#endif
   CRNLIB_ASSERT(maximumCount >= initialCount);
 #if !defined(__APPLE__)
-  maximumCount, pName;

```

Comment:
Using `(void)x` seems nicer here instead of writing the function declaration twice

Reply:
Now done, also done on the other occurrences of duplicated declaration you reported.